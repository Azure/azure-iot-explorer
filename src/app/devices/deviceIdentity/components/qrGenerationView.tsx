/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from '@fluentui/react';
import QRCode from 'react-qr-code';
import { CollapsibleSection } from '../../../shared/components/collapsibleSection';
import '../../../css/_qrGeneration.scss';
import { ResourceKeys } from '../../../../localization/resourceKeys';

const QR_SIZE = 200;

export interface QrGenerationDataProps {
    hostName: string;
    deviceId: string;
    deviceKey: string;
}
export const QrGenerationView: React.FC<QrGenerationDataProps> = (props: QrGenerationDataProps) => {
    const { t } = useTranslation();
    const { deviceId, deviceKey, hostName } = props;
    const connection = Buffer.from(JSON.stringify({
        deviceId,
        deviceKey,
        hostName
    })).toString('base64');

    return (
        <CollapsibleSection
            expanded={false}
            label={t(ResourceKeys.deviceIdentity.qrCode.label)}
            tooltipText={t(ResourceKeys.deviceIdentity.qrCode.toolTip)}
        >
            <div className="qrCodeGeneration">
                <h1>{t(ResourceKeys.deviceIdentity.qrCode.headerText)}</h1>
                <p>{t(ResourceKeys.deviceIdentity.qrCode.callToAction)}</p>
                <section className="storeLinks">
                    <Link
                        href="https://play.google.com/store/apps/details?id=com.iot_pnp"
                        target="_blank"
                    >
                        <img src="images/qrCodes/playstore.png" />
                    </Link>
                    <Link
                        href="https://apps.apple.com/app/iot-plug-and-play/id1563783687"
                        target="_blank"
                    >
                        <img src="images/qrCodes/appstore.png" />
                    </Link>
                </section>
                <section>
                    <div className="qrCode">
                        <QRCode value={connection} size={QR_SIZE} />
                    </div>
                </section>
            </div>
        </CollapsibleSection>
    );
};
