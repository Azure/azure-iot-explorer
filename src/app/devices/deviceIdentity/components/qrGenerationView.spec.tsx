/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
 import * as React from 'react';
 import { shallow } from 'enzyme';
 import { QrGenerationView } from './qrGenerationView';

 describe('DeviceModules', () => {
     it('matches snapshot', () => {
         expect(shallow(
             <QrGenerationView
                deviceId="some_device"
                deviceKey="invalid_device_key"
                hostName="invalid_host"
             />
         )).toMatchSnapshot();
     });
 });
