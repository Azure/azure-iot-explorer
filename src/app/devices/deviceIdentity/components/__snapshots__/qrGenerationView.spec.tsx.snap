// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceModules matches snapshot 1`] = `
<CollapsibleSection
  expanded={false}
  label="deviceIdentity.qrCode.label"
  tooltipText="deviceIdentity.qrCode.toolTip"
>
  <div
    className="qrCodeGeneration"
  >
    <h1>
      deviceIdentity.qrCode.headerText
    </h1>
    <p>
      deviceIdentity.qrCode.callToAction
    </p>
    <section
      className="storeLinks"
    >
      <StyledLinkBase
        href="https://play.google.com/store/apps/details?id=com.iot_pnp"
        target="_blank"
      >
        <img
          src="images/qrCodes/playstore.png"
        />
      </StyledLinkBase>
      <StyledLinkBase
        href="https://apps.apple.com/app/iot-plug-and-play/id1563783687"
        target="_blank"
      >
        <img
          src="images/qrCodes/appstore.png"
        />
      </StyledLinkBase>
    </section>
    <section>
      <div
        className="qrCode"
      >
        <Memo(QRCode)
          size={200}
          value="eyJkZXZpY2VJZCI6InNvbWVfZGV2aWNlIiwiZGV2aWNlS2V5IjoiaW52YWxpZF9kZXZpY2Vfa2V5IiwiaG9zdE5hbWUiOiJpbnZhbGlkX2hvc3QifQ=="
        />
      </div>
    </section>
  </div>
</CollapsibleSection>
`;
